docs/.vuepress/config.js

```diff
@@ -72,20 +72,47 @@ module.exports = {
             text: 'Config Reference',
             link: '/config/'
           },
-          {
-            text: 'Plugin Dev Guide',
-            items: [
-              { text: 'Plugin Dev Guide', link: '/dev-guide/plugin-dev.md' },
-              { text: 'UI Plugin Info', link: '/dev-guide/ui-info.md' },
-              { text: 'UI Plugin API', link: '/dev-guide/ui-api.md' },
-              { text: 'UI Localization', link: '/dev-guide/ui-localization.md' }
-            ]
-          },
           {
             text: 'Plugins',
             items: [
-              { text: 'Core plugins', link: '/core-plugins/' },
-              { text: 'Browse plugins', link: 'https://awesomejs.dev/for/vue-cli/' }
+              {
+                text: 'Configuration',
+                items:
+                [
+                  { text: 'Core', link: '/core-plugins/' },
+                ]
+              },
+              {
+                text: 'Discover',
+                items:
+                [
+                  { text: 'Browse', link: 'https://awesomejs.dev/for/vue-cli/' }
+                ]
+              },
+              {
+                text: 'Development',
+                items:
+                [
+                  { text: 'Guide', link: '/dev-guide/plugin-dev.md' },
+                ]
+              },
+              {
+                text: 'API',
+                items:
+                [
+                  { text: 'Plugin', link: '/dev-guide/plugin-api.md' },
+                  { text: 'Generator', link: '/dev-guide/generator-api.md' },
+                ]
+              },
+              {
+                text: 'UI',
+                items:
+                [
+                  { text: 'Info', link: '/dev-guide/ui-info.md' },
+                  { text: 'API', link: '/dev-guide/ui-api.md' },
```

Comment:
Hopefully this duplicate label here won't create any confusion given there's a section label above in https://github.com/vuejs/vue-cli/pull/5655/files#diff-d669c10d59b56bb12a815dd48ce8d884R100.

Maybe it's worth considering refactoring these closely related documentation sections. 🔮